src/loaders/zarrLoader.js

```diff
@@ -0,0 +1,113 @@
+import { range } from '../layers/microscopy-viewer-layer/utils';
+
+export default class ZarrLoader {
+  constructor(data, isRgb, scale, dimNames) {
+    let base;
+    if (Array.isArray(data)) {
+      this.isPyramid = true;
+      this._data = data;
+      [base] = data;
+    } else {
+      this.isPyramid = false;
+      base = data;
+    }
```

Comment:
I think this is fine, but it might be going in a direction where subclassing would be appropriate, particularly if the behavior in the two cased diverges more.

Reply:
I think I agree with Chuck here, but might go further  (and maybe we should discuss tomorrow in-person).  Maybe we can have a pyramid parent class and a static image parent class - it is possible to have static TIFF images and it would be nice to support it.  In fact, as I type this, I realize that there is no reason why even the CODEX data could not just be left in its `tiff` format instead of having to worry about a convertor, and I can write a more direct accessor with `geotiff`.  This would motivate the separation even more.

Reply:
I had an initial draft with this using composition where the `ZarrPyramidLoader` contained an array of `ZarrLoaders`. I ended up settling on these because I didn't think instantiating a lot of `ZarrLoaders` was necessary. We should chat tomorrow at standup.

Reply:
I think I'm going to push for keeping the `isPyramid` here. The only thing that is different between the two classes is the notion of what level of the pyramid you are on. All the stuff regarding how to index the zarr array (at each level is the same).